utils/releaser/src/utils.rs

```diff
@@ -69,3 +70,72 @@ pub fn root() -> PathBuf {
     }
     project_root
 }
+
+pub fn determine_new_version(bump_type: Option<String>) -> Option<String> {
+    let mut current_version: Vec<i32> = core_version()
+        .split('.')
+        .map(|f| f.parse().unwrap())
+        .collect();
+
+    let bump_type = bump_type.unwrap_or_else(|| "patch".to_string());
+    match bump_type.as_ref() {
+        "major" => current_version[0] += 1,
+        "minor" => current_version[1] += 1,
+        "patch" => current_version[2] += 1,
+        _ => panic!(
+            "{} is an undefined version bump. accepted values are: major, minor, patch",
+            bump_type
+        ),
+    }
+
+    let new_version = current_version
+        .iter()
+        .map(|f| f.to_string())
+        .collect::<Vec<String>>()
+        .join(".");
+
+    Some(new_version)
+}
+
+pub fn edit_cargo_version(cargo_path: &str, version: &str) {
+    let mut server = fs::read_to_string([cargo_path, "Cargo.toml"].join(""))
+        .unwrap()
+        .parse::<Document>()
+        .unwrap();
+
+    server["package"]["version"] = value(version);
+
+    fs::write(cargo_path, server.to_string()).unwrap();
+}
+
+pub fn bump_versions(bump_type: Option<String>) {
+    let new_version = determine_new_version(bump_type).unwrap_or_else(core_version);
+    let new_version = new_version.as_str();
+
+    let cargos_to_update = vec![
+        "/clients/admin/",
+        "/clients/cli/",
+        "/clients/egui/",
+        "/server/server/",
+        "/libs/core/",
+    ];
+    for cargo_path in cargos_to_update {
+        edit_cargo_version(cargo_path, new_version);
+    }
+
+    //apple
+    //TODO figure out what sbuild number and appropriate plist
+    Command::new("/usr/libexec/Plistbuddy").args([
```

Comment:
<img width="327" alt="image" src="https://user-images.githubusercontent.com/821972/215834632-a26ee7df-354c-45c6-be5e-dc922985b12f.png">

this are the info files you want to modify

`CFBundleShortVersionString` will get the core version

`CFBundleVersion` actually will never be updated I'm pretty sure